code/datums/enumeration/enumeration_operations.dm

```diff
@@ -0,0 +1,16 @@
+
+///Kinda wish we had static classes here
+
+/**
+ * Allows performing some operation over each element of another enumerator.
+ * At every step of the enumeration, the callback will be called with the input as the result
+ * and then the result of that callback will be returned instead of the initial value passed
+ * from the original enumerator.
+ * Returns an enumerator.
+ */
+/datum/enumerator/proc/foreach(datum/callback/callback)
+	return new /datum/enumerator/callback(src, callback)
```

Comment:
you could use defines instead of creating this wrapper

Reply:
I could, but I want them to specifically be a proc on /datum/enumerator rather than something that is globally accessible with 2 parameters.